lib/chef/resource/chocolatey_source.rb

```diff
@@ -31,23 +32,34 @@ class ChocolateySource < Chef::Resource
       property :bypass_proxy, [TrueClass, FalseClass], default: false,
                description: "Whether or not to bypass the system's proxy settings to access the source."
 
+      property :admin_only, [TrueClass, FalseClass], default: false,
+               description: "Whether or not to set the source to be accessible to only admins."
+
+      property :allow_self_service, [TrueClass, FalseClass], default: false,
+               description: "Whether or not to set the source to be used for self service."
+
       property :priority, Integer, default: 0,
                description: "The priority level of the source."
 
+      property :source_state, [TrueClass, FalseClass], default: false, desired_state: false, skip_docs: true
+
       load_current_value do
         element = fetch_source_element(source_name)
         current_value_does_not_exist! if element.nil?
 
         source_name element["id"]
         source element["value"]
         bypass_proxy element["bypassProxy"] == "true"
+        admin_only element["adminOnly"] == "true"
+        allow_self_service element["selfService"] == "true"
         priority element["priority"].to_i
+        source_state element["disabled"] == "true"
       end
 
       # @param [String] id the source name
       # @return [REXML::Attributes] finds the source element with the
       def fetch_source_element(id)
-        require "rexml/document" unless defined?(REXML::Document)
+        require "rexml/document"
```

Comment:
that needs to get backed out, the idempotent behavior of `require` in ruby is poor and needs the guard

Reply:
I don't recall removing that 😢 I have added this back in.